version: "3.1"
services:
  psql:
    image: postgres
    # restart: always
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_DB=issues
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=pass
    volumes:
      - pgdata:/var/lib/postgresql/data 

  mongo:
    container_name: mongo
    image: mongo
    # restart: always
    ports:
      - "27017:27017"
    volumes:
      - ./data:/data/db
    networks:
      - maestro_traefik

  mongo-express:
    image: mongo-express
    # restart: always
    ports:
      - "8081:8081"
    networks:
      - maestro_traefik

  issues-db-api:
    container_name: issues-db-api
    build: ./issues-db-api/
    image: issues-db-api
    ports:
      - "8000:8000"
    networks:
      - maestro_traefik
    environment:
      - MONGO_URL=mongodb://mongo:27017
      - DOCKER=True
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.issues-db-api.rule=Host(`maestro.localhost`)"
      - "traefik.http.routers.issues-db-api.rule=PathPrefix(`/issues-db-api`)"
      - "traefik.http.routers.issues-db-api.entrypoints=websecure"
      - "traefik.http.routers.issues-db-api.tls=true"
      - "traefik.http.routers.issues-db-api.middlewares=issues-db-api"
      - "traefik.http.middlewares.issues-db-api.stripprefix.prefixes=/issues-db-api"

  backup-issues-db:
    container_name: backup-issues-db
    build: ./backup/
    image: backup-issues-db
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

volumes:
  pgdata:
networks:
  maestro_traefik:
    external: true
